include/onnxruntime/core/framework/sparse_tensor.h

```diff
@@ -1,72 +1,317 @@
-#pragma once
-
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+#pragma once
+
 #include "core/framework/data_types.h"
 #include "core/framework/tensor_shape.h"
 #include "core/framework/tensor.h"
 
 namespace onnxruntime {
 
+class IDataTransfer;
+class DataTransferManager;
+
+/**
+ * @brief This is a Sparse Format enumeration representing bitflags
+ * 
+ * 
+ */
+enum class SparseFormatFlags : uint32_t {
+  kUndefined = 0x0U,        // For completeness
+  kCoo = 0x1U,              // 1-D or 2-D indices
+  kCsrc = 0x1U << 1,        // Both CSR(C)
+  kBlockSparse = 0x1U << 2  // as in OpenAI
+};
+
+std::ostream& operator<<(std::ostream&, SparseFormatFlags);
+
+inline SparseFormatFlags operator|(SparseFormatFlags flags, SparseFormatFlags flag) noexcept {
+  return static_cast<SparseFormatFlags>(static_cast<uint32_t>(flags) | static_cast<uint32_t>(flag));
+}
+
+inline SparseFormatFlags operator&(SparseFormatFlags flags, SparseFormatFlags flag) noexcept {
+  return static_cast<SparseFormatFlags>(static_cast<uint32_t>(flags) & static_cast<uint32_t>(flag));
+}
+
+inline SparseFormatFlags TurnOff(SparseFormatFlags flags, SparseFormatFlags flag) noexcept {
+  return static_cast<SparseFormatFlags>(static_cast<uint32_t>(flags) & ~static_cast<uint32_t>(flag));
+}
+
```

Comment:
nit: I would have expected the formats to be exclusive (so no concept of 'turn off' and 'set' ensures only one format is selected). is that not the case?

Reply:
Yes, the formats are mutually exclusive. However, the formats sometimes vary depending on the library such as CUDA and we need to sometimes move back and forth. So flags were meant as additional annotations. I will get rid of that.

Reply:
Formats are no longer flags.